File: HRMS/HRMS.Web/Modules/ServerTypes/Operations/LeaveApprovalColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { LeaveApprovalRow } from "./LeaveApprovalRow";
import { LeaveStatus } from "./LeaveStatus";

export interface LeaveApprovalColumns {
    ApprovalId: Column<LeaveApprovalRow>;
    LeaveApplicationNo: Column<LeaveApprovalRow>;
    ApproverUsername: Column<LeaveApprovalRow>;
    ApprovalLevel: Column<LeaveApprovalRow>;
    ApprovalDate: Column<LeaveApprovalRow>;
    Status: Column<LeaveApprovalRow>;
    Remarks: Column<LeaveApprovalRow>;
}

export class LeaveApprovalColumns extends ColumnsBase<LeaveApprovalRow> {
    static readonly columnsKey = 'Operations.LeaveApproval';
    static readonly Fields = fieldsProxy<LeaveApprovalColumns>();
}

[LeaveStatus]; // referenced types

File: HRMS/HRMS.Web/Modules/Operations/LeaveApproval/LeaveApprovalColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace HRMS.Operations.Columns;

[ColumnsScript("Operations.LeaveApproval")]
[BasedOnRow(typeof(LeaveApprovalRow), CheckNames = true)]
public class LeaveApprovalColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int ApprovalId { get; set; }
    [EditLink]
    public string LeaveApplicationNo { get; set; }
    public string ApproverUsername { get; set; }
    public int ApprovalLevel { get; set; }
    public DateTime ApprovalDate { get; set; }
    public LeaveStatus Status { get; set; }
    public string Remarks { get; set; }
}
